setuptools/__init__.pyi

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Mapping, Optional, Sequence, Union
+
+import setuptools
+
+
+def setup(
+    name: str,
+    version: Optional[str] = None,
```

Comment:
FYI, version is allowed to be a Number (float). That usage is discouraged, but hasn't yet been deprecated. Definitely don't want to advertise that fact, though, so perhaps this approach is best and violators can use `type:ignore`.